### PR TITLE
OSDOCS-10287 OCP 4.15.10 RNs

### PR DESCRIPTION
**Pull request by cbippley**
Version(s): 4.15

Issue: https://issues.redhat.com/browse/OSDOCS-10287

Link to docs preview: [Preview](https://75108--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-15-release-notes.html#ocp-4-15-10)

QE review: Not needed
- [ ] QE has approved this change.

Additional information: Expected merge day April 25. Errata links will not work until then. EDIT: Had to change the release date due to a delay.

### CODE REVIEW

**Review by mletalie — COMMENTED** [review]
/label peer-review-in-progress

File: release_notes/ocp-4-15-release-notes.adoc [hunk 1]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
 For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
 ====
 
-// 4.15.9
+[id="ocp-4-15-10"]
+=== RHSA-2024:1887 - {product-title} 4.15.10 bug fix and security update
+
+Issued: 2024-04-25
+
+{product-title} release 4.15.10, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:1887[RHSA-2024:1887] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:1892[RHSA-2024:1892] advisory.
```
Comment (mletalie) [inline release_notes/ocp-4-15-release-notes.adoc:2724]: link:https://access.redhat.com/errata/RHSA-2024:1892[RHSA-2024:1892] not currently working, but I do realize it just may not be live yet....

File: release_notes/ocp-4-15-release-notes.adoc [hunk 2]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+[source,terminal]
+----
+$ oc adm release info 4.15.10 --pullspecs
+----
+
+[id="ocp-4-15-10-bug-fixes"]
+==== Bug fixes
+
+* Previously, Clusters born before 4.7 will had signer keys for api-int endpoint updated unexpectedly when users upgraded to OpenShift Container Platform 4.15 due to the installer deleting the SecretTypeTLS and then recreating the secret with `kubernetes.io/tls` type.  With this release, the issue is resloved by the installer changing the secret type without deleting the secret. (link:https://issues.redhat.com/browse/OCPBUGS-31807[*OCPBUGS-31807*])
```
Comment (mletalie) [inline release_notes/ocp-4-15-release-notes.adoc:2738]:
```suggestion
* Previously, Clusters created before 4.7 will have had signer keys for api-int endpoint updated unexpectedly when users upgraded to OpenShift Container Platform 4.15 due to the installer deleting the SecretTypeTLS and then recreating the secret with `kubernetes.io/tls` type.  With this release, the issue is resolved by the installer changing the secret type without deleting the secret. (link:https://issues.redhat.com/browse/OCPBUGS-31807[*OCPBUGS-31807*])
```

File: release_notes/ocp-4-15-release-notes.adoc [hunk 3]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously,  when users imported image stream tags, ICSP (In-Circuit Serial Programming) and IDMS (Integrated Database Managing System) were not allowed to co-exist. OpenShift ignored any IDMS/ITMS created by the user and favored ICSP. With this release, they are allowed to co-exist so importing image stream tags will now respect IDMS/ITMS when ICSP is also present. (link:https://issues.redhat.com/browse/OCPBUGS-31469[*OCPBUGS-31469*])
```
Comment (mletalie) [inline release_notes/ocp-4-15-release-notes.adoc:2740]:
```suggestion
* Previously,  when users imported image stream tags, ICSP (In-Circuit Serial Programming) was not allowed to co-exist with IDMS (Integrated Database Managing System) and ImageTagMirrorSet (ITMS). OpenShift ignored any IDMS/ITMS created by the user and favored ICSP. With this release, they are allowed to co-exist as importing image stream tags will now respect IDMS/ITMS when ICSP is also present. (link:https://issues.redhat.com/browse/OCPBUGS-31469[*OCPBUGS-31469*])
```

Comment (mletalie) [reply release_notes/ocp-4-15-release-notes.adoc:2740]: Not sure if this is the right way to go about it (my suggestion), but ITMS is not addressed specifically. Not sure if it needs to be?

File: release_notes/ocp-4-15-release-notes.adoc [hunk 4]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously, Terraform would create the compute server group with the policy set for the control plane. As a consequence, the 'serverGroupPolicy' property of `install-config.yaml` was ignored for the Computes. With this release, the server group policy set in `install-config.yaml` for the compute MachinePool is correctly applied at install-time in the Terraform flow.  (link:https://issues.redhat.com/browse/OCPBUGS-31335[*OCPBUGS-31335*])
```
Comment (mletalie) [inline release_notes/ocp-4-15-release-notes.adoc:2742]: Should "Computes" be capitalized?

File: release_notes/ocp-4-15-release-notes.adoc [hunk 5]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously, Projects that specified a non-intersecting openshift.io/node-selector project selector with pods `.spec.nodeName` could cause runaway Pod creation in Deployments. With this release, pods with non-intersecting `.spec.nodeName` will not be admitted by the API server which resolves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-29922[*OCPBUGS-29922*])
+
+* Previously, a remote attacker with basic login credentials could check the pod manigest to discover a repository pull secret. With this release, the vulnerability has been fixed. (link:https://issues.redhat.com/browse/OCPBUGS-28769[*OCPBUGS-28769*])
```
Comment (mletalie) [inline release_notes/ocp-4-15-release-notes.adoc:2746]:
```suggestion
* Previously, a remote attacker with basic login credentials could check the pod manifest to discover a repository pull secret. With this release, the vulnerability has been fixed. (link:https://issues.redhat.com/browse/OCPBUGS-28769[*OCPBUGS-28769*])
```

**Review by mletalie — COMMENTED** [review]
/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done
/lgtm

**Comment by mletalie** [comment]
Hello @cbippley,
   Had a few comments/suggestions, otherwise LGTM! Thanks!

File: release_notes/ocp-4-15-release-notes.adoc (hunk 1 above)
Comment (cbippley) [reply release_notes/ocp-4-15-release-notes.adoc:2724]: It's not live yet. Thanks for checking!

File: release_notes/ocp-4-15-release-notes.adoc [hunk 6]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+[source,terminal]
+----
+$ oc adm release info 4.15.10 --pullspecs
+----
+
+[id="ocp-4-15-10-bug-fixes"]
+==== Bug fixes
+
+* Previously, Clusters created before OpenShift Container Platform 4.7 will have had signer keys for api-int endpoint updated unexpectedly when users upgraded to OpenShift Container Platform 4.15 due to the installer deleting the SecretTypeTLS and then recreating the secret with `kubernetes.io/tls` type.  With this release, the issue is resolved by the installer changing the secret type without deleting the secret. (link:https://issues.redhat.com/browse/OCPBUGS-31807[*OCPBUGS-31807*])
```
Comment (stevsmit) [inline release_notes/ocp-4-15-release-notes.adoc:2738]:
```suggestion
* Previously, clusters created before {product-title} 4.7 will have had signer keys for api-int endpoint updated unexpectedly when users upgraded to {product-title} 4.15 due to the installer deleting the SecretTypeTLS and then recreating the secret with `kubernetes.io/tls` type.  With this release, the issue is resolved by the installer changing the secret type without deleting the secret. (link:https://issues.redhat.com/browse/OCPBUGS-31807[*OCPBUGS-31807*])
```

File: release_notes/ocp-4-15-release-notes.adoc [hunk 7]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously, when users imported image stream tags, ICSP (In-Circuit Serial Programming) was not allowed to co-exist with IDMS (Integrated Database Managing System) and ImageTagMirrorSet (ITMS). OpenShift ignored any IDMS/ITMS created by the user and favored ICSP. With this release, they are allowed to co-exist as importing image stream tags will now respect IDMS/ITMS when ICSP is also present. (link:https://issues.redhat.com/browse/OCPBUGS-31469[*OCPBUGS-31469*])
```
Comment (stevsmit) [inline release_notes/ocp-4-15-release-notes.adoc:2740]: ICSP (In-Circuit Serial Programming)
IDMS (Integrated Database Managing System)
ImageTagMirrorSet (ITMS)

Suggesting parallelism, unless there's a reason you didn't want to: 

```suggestion
* Previously, when users imported image stream tags, ICSP (In-Circuit Serial Programming) was not allowed to co-exist with IDMS (Integrated Database Managing System) and  ITMS (ImageTagMirrorSet). OpenShift ignored any IDMS/ITMS created by the user and favored ICSP. With this release, they are allowed to co-exist as importing image stream tags will now respect IDMS/ITMS when ICSP is also present. (link:https://issues.redhat.com/browse/OCPBUGS-31469[*OCPBUGS-31469*])
```

File: release_notes/ocp-4-15-release-notes.adoc [hunk 8]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously, Terraform would create the compute server group with the policy set for the control plane. As a consequence, the 'serverGroupPolicy' property of `install-config.yaml` was ignored for the compute server group. With this release, the server group policy set in `install-config.yaml` for the compute MachinePool is correctly applied at install-time in the Terraform flow.  (link:https://issues.redhat.com/browse/OCPBUGS-31335[*OCPBUGS-31335*])
```
Comment (stevsmit) [inline release_notes/ocp-4-15-release-notes.adoc:2742]: I think generally we're supposed to follow `.yaml` with a noun:

```suggestion
* Previously, Terraform would create the compute server group with the policy set for the control plane. As a consequence, the 'serverGroupPolicy' property of the `install-config.yaml` file was ignored for the compute server group. With this release, the server group policy set in the `install-config.yaml` file for the compute MachinePool is correctly applied at install-time in the Terraform flow.  (link:https://issues.redhat.com/browse/OCPBUGS-31335[*OCPBUGS-31335*])
```

File: release_notes/ocp-4-15-release-notes.adoc [hunk 9]
```diff
@@ -2716,7 +2716,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously, Projects that specified a non-intersecting openshift.io/node-selector project selector with pods `.spec.nodeName` could cause runaway Pod creation in Deployments. With this release, pods with non-intersecting `.spec.nodeName` will not be admitted by the API server which resolves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-29922[*OCPBUGS-29922*])
```
Comment (stevsmit) [inline release_notes/ocp-4-15-release-notes.adoc:2744]:
```suggestion
* Previously, projects that specified a non-intersecting openshift.io/node-selector project selector with pods `.spec.nodeName` could cause runaway Pod creation in Deployments. With this release, pods with non-intersecting `.spec.nodeName` are not admitted by the API server which resolves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-29922[*OCPBUGS-29922*])
```

**Comment by stevsmit** [comment]
@cbippley a few nits before merging

**Comment by cbippley** [comment]
Thanks @stevsmit! Changes made, ready for merge once it builds.

**Pull request merged**